pkg/cri/opts/utils_windows.go

```diff
@@ -0,0 +1,76 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package opts
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/containerd/containerd/log"
+	"github.com/containerd/containerd/mount"
+	"github.com/pkg/errors"
+)
+
+func copyFromVolumes(ctx context.Context, volumeMounts map[string]string, mounts []mount.Mount) (err error) {
+	root, err := ioutil.TempDir("", "ctd-volume")
+	if err != nil {
+		return err
+	}
+	// We change RemoveAll to Remove so that we either leak a temp dir
+	// if it fails but not RM snapshot data.
+	// refer to https://github.com/containerd/containerd/pull/1868
+	// https://github.com/containerd/containerd/pull/1785
+	defer os.Remove(root) // nolint: errcheck
+	for _, m := range mounts {
+		// appending the layerID to the root.
+		if err := m.Mount(filepath.Join(root, filepath.Base(m.Source))); err != nil {
+			return err
+		}
+
+		defer func(mm mount.Mount) {
+			if uerr := mount.Unmount(mm.Source, 0); uerr != nil {
+				log.G(ctx).WithError(uerr).Errorf("Failed to unmount snapshot %q", mm.Source)
+				if err == nil {
+					err = uerr
+				}
+			}
+		}(m)
+	}
+
+	for host, volume := range volumeMounts {
+		// The volume may have been defined with a C: prefix, which we can't use here.
+		volume = strings.TrimPrefix(volume, "C:")
```

Comment:
similar to @kzys's comment, if this is the only difference between non-Windows and Windows impl it might make more sense to just switch on platform for this one statement? Hard to tell if there are other differences I'm missing.

Reply:
Done